cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -199,6 +201,36 @@ func generateInitramfsMounts() (err error) {
 	return nil
 }
 
+func canInstallAndRun(mst *initramfsMountsState) (bool, error) {
+	currentSeed, err := seed.Open(boot.InitramfsUbuntuSeedDir, mst.recoverySystem)
+	if err != nil {
+		return false, err
+	}
+	preseedSeed, ok := currentSeed.(seed.PreseedCapable)
+	if !ok {
+		return false, nil
+	}
+
+	if !preseedSeed.HasArtifact("preseed.tgz") {
+		return false, nil
+	}
+
+	// If kernel has fde-setup hook, then we should also have fde-setup in initramfs
+	kernelPath := filepath.Join(boot.InitramfsRunMntDir, "kernel")
+	gadgetHasFdeSetup := osutil.FileExists(filepath.Join(kernelPath, "meta", "hooks", "fde-setup"))
```

Comment:
s/gadget/kernel/ here?

Reply:
Done in c5a44c0c1b98e1301786e6567898951bbbeebdfc